### PR TITLE
[Enhancement] Do not print password of failed login user

### PR DESCRIPTION
**Pull request by gengjun-git**
## What type of PR is this：
- [ ] BugFix
- [ ] Feature
- [x] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Which issues of this PR fixes ：

Fixes #

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr will affect users' behaviors
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto backported to target branch
  - [x] 3.0
  - [x] 2.5
  - [x] 2.4
  - [x] 2.3

### CODE REVIEW

File: fe/fe-core/src/main/java/com/starrocks/mysql/security/LdapSecurity.java [hunk 1]
```diff
@@ -47,7 +47,8 @@ public static boolean checkPassword(String dn, String password) {
             ctx = new InitialDirContext(env);
             return true;
         } catch (Exception e) {
-            LOG.warn("check ldap password failed, dn = {}, password = {}", dn, password, e);
+            LOG.warn("check ldap password failed, dn = {}", dn, e);
+            LOG.debug("check ldap password failed, dn = {}, password = {}", dn, password);
```
Comment (nshangyiming) [inline fe/fe-core/src/main/java/com/starrocks/mysql/security/LdapSecurity.java:51]: why still log the password? is it necessary even in debug log level?

Comment (imay) [reply fe/fe-core/src/main/java/com/starrocks/mysql/security/LdapSecurity.java:51]: I think we can't print credential information in the log.

**Review by HangyuanLiu — APPROVED** [review]

**Review by imay — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
@Mergifyio backport branch-3.0

**Comment by wanpengfei-git** [comment]
@Mergifyio backport branch-2.4

**Comment by wanpengfei-git** [comment]
@Mergifyio backport branch-2.3

**Pull request merged**